src/Patient.java

```diff
@@ -0,0 +1,128 @@
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class Patient {
+	
+	private String name;
+	private ArrayList<Headache> headacheList;
+	
+	public Patient(){
+		headacheList = new ArrayList<Headache>();
+	}
+	
+	public String getName() {
+		return name;
+	}
+	public void setName(String name) {
+		this.name = name;
+	}
+	
+	public void addHeadache(Headache headache){
+		headacheList.add(headache);
+	}
+	
+	public String paitentData(){
+		String sum=null;
+		for(int i=0;i<headacheList.size();i++){
+			sum="\nDate = "+headacheList.get(i).getDate().toString()+
+					"\nDuration = "+headacheList.get(i).getDuration()+
+					"\nType = "+headacheList.get(i).getType()+
+					"\nSymptoms = "+headacheList.get(i).getProdromeSymptoms()+" "+headacheList.get(i).getConcurrentSymptoms()+" "+headacheList.get(i).getPostdromeSymptoms()+
+					"\nSeverity = "+headacheList.get(i).getPainDescriptor().getSeverity();
+			
+					for(PainDescriptorContainer container : headacheList.get(i).getPainDescriptor().getLocationList())
```

Comment:
NPE is thrown at this line if this method is executed from Tracker.  This was tested to fulfill an aspect of the grading rubric, "raw data output".  Please also see note on grading rubric.